### DISCUSSION TITLE
addNodeBeforeNode is deprecated - What is the new way?

### DISCUSSION THREAD

**Discussion by Mirko-Tebaldi** [rectorphp/rector#7790] [Q&A] [answered]
I am using a lot the `addNodeBeforeNode` method of `NodesToAddCollector`

See this: vendor/rector/rector/packages/PostRector/Collector/NodesToAddCollector.php 

Actually the `addNodeBeforeNode`  is deprecated.

Which is the current suggested way to prepend a node to another?

**Comment by TomasVotruba** [comment, marked as answer]
Hi, if you want to replace 1 node with 2, just return array of 2 stmts:

```php
return [$firstStmt, $secondStmt];
```

That will update the node tree with these 2 new nodes.
